honeybee_grasshopper_radiance/src/HB Extract HDR.py

```
@@ -0,0 +1,218 @@
+    valid_view = False
+    
+    # read hdr image and search for a valid view
+    with open(hdr_path, 'r') as hdr_file:
+        for lineCount, line in enumerate(hdr_file):
+            if lineCount < 200:
+                low_line = line.lower()
+                if not low_line.startswith('\t'):
+                    if low_line.startswith('view='):
+                        valid_view = True
+                        if line.find('-vth') > -1:
+                            projection = '-vth'
+                        elif line.find('-vta') > -1:
+                            projection = '-vta'
+                        hdr_view = View.from_string('hdr_view', line)
+            else:  # no need to check the rest of the document
+                break
+    if not valid_view:
+        raise ValueError(
+            'Connected _hdr image does not contain a valid view in the header.\n'
+            'Note that indented views in the header will be ignored by pinterp.')
+    if not hdr_view.type in ('a', 'h'):
+        msg = 'Expected view type -vta or -vth in _hdr. Got view type -vt{}.'
+        raise ValueError(msg.format(hdr_view.type))
+    return hdr_view
+
+
+def check_view_points(view, hdr_view):
+    """Check if view points of output view and input HDR are matching.
+    
+    A ValueError is raised if the view points are not matching.
+    
+    Args:
+        view: A Honeybee Radiance View to extract.
+        hdr_view: A Honeybee Radiance View from the input HDR.
+    """
+    if not view.position == hdr_view.position:
+        msg = 'View points of _view and _hdr are not matching.\n' \
+        'Got _view = {} and _hdr = {}.'
+        raise ValueError(msg.format(view.position, hdr_view.position))
+
+
+def check_resolution(hdr_path, resolution, view, hdr_view):
+    """Check the resolution of the output HDR as well as the input HDR.
+    
+    A warning is raised if the HDR image dimensions are not square. A warning is
```

Am I correct that the Radiance command still succeeds even when the output resolution is larger than the input resolution? If so, this is good. Otherwise, it would be good to raise an error here instead of a warning.

---

Yes, that is correct.